### ISSUE TITLE
Database cleanup (indexes, constraints).

### ISSUE THREAD

**Issue by Hinton** [modulargaming/modulargaming#142]
We should go over the database and clean it up.

There are a few models that contain custom delete code (forum poll, category). Perhaps this code could be moved to the database with constraints?

We should also add more indexes and constraints for relationships to keep the database clean from "dead" data.

**Comment by curtisdelicata** [comment]
I agree that we should remove custom delete code and use constraints instead. Do you have any ideas @Lewis @happyDemon @jannesiera

**Comment by Hinton** [comment]
We need to decide on a naming standard. Our current system is kinda messy and the naming makes no sense.

**Tables:**
Plural table names.

**Indexes:**
[Column name] with prefixes, uniq_ (Unique), fk_ (Foreign Key)

**Constraints:**
Kohana uses [table name]_ibfk_1

Should we follow Kohanas convention or modify it?

**Comment by curtisdelicata** [comment]
We should follow Kohana conventions? Up to you.

**Comment by curtisdelicata** [comment]
Can this be closed now?

**Comment by Hinton** [comment]
No, we still have more indexes and constraints needed, but its not critical for the preview release or 1.0.

**Comment by curtisdelicata** [comment]
Thanks for your reply. No problem.
